flink-ml-iteration/src/main/java/org/apache/flink/iteration/datacache/nonkeyed/Segment.java

```diff
@@ -63,16 +98,18 @@ public boolean equals(Object o) {
         }
 
         Segment segment = (Segment) o;
-        return count == segment.count && size == segment.size && Objects.equals(path, segment.path);
+        return count == segment.count
+                && fsSize == segment.fsSize
```

Comment:
Should a segment be uniquely identified by its count and path only?

I suppose if we write segment data from memory to file, it should still represent the same segment, right?

Same for `hashCode()`.